Switch to new feed without BlogMonster

Cheesebaron opened the pull request:
We have a hacked together version of combined feed currently, which primarily uses blog monster to combine the downloaded feeds and for some caching of feeds for an hour.

This PR decouples us from BlogMonster but aslo adds a simple retry mechanism using a Polly policy. Also instead of the caching mechanism BlogMonster provides, we are now using Polly to help us with that, which simplifies the code significantly.
Also it seems like the BlogMonster project is pretty much dead, so it seems fit to do something about that.

TODO:
- [x] Fix Test suite

Fixes #230
Helps us getting along with #536 and #588 when we switch to ASP.NET Core

Cheesebaron commented:
@planetxamarin/owners can I get a couple of reviews on this one? I am very confident it works. However, would love some eyes on this.

In general, the feed loads in around 15 or 16 seconds now if all feeds respond in a timely manner. A huge improvement over our current Blog Monster based feed, which can take up to a minute.

nickrandolph commented:
I'll try to take a look over the weekend. Do you know how much effort it would be to move to new project system as well?

Cheesebaron commented:
@icebeam7 where is it hosted, it often takes more than 10-15 seconds to reach it? This is around the amount of time it normally takes to fetch _all_ the 100+ feeds we have!

jfversluis commented:
@icebeam7 I see the same thing. Over 20 seconds even to load your regular website.

![image](https://user-images.githubusercontent.com/939291/60336998-21109f80-99a2-11e9-8dd1-450029cc96c9.png)

nickrandolph commented on src/Firehose.Web/Infrastructure/NewCombinedFeedSource.cs:
```diff
@@ -0,0 +1,228 @@
+                    new ProductInfoHeaderValue("PlanetXamarin", $"{GetType().Assembly.GetName().Version}"));
+                httpClient.Timeout = TimeSpan.FromSeconds(10);
+
+                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11;
```
should this be changed to remove tls11 and add tls13?

Cheesebaron replied:
Yeah, I guess that would be the better practice.

Cheesebaron replied:
TSL13 doesn't seem to be available in full framework

nickrandolph commented on src/Firehose.Web/Firehose.Web.csproj:
```diff
@@ -196,7 +196,6 @@
     <Compile Include="App_Start\RouteConfig.cs" />
     <Compile Include="Authors\AllanRitchie.cs" />
     <Compile Include="Authors\AndrewHoefling.cs" />
-    <Compile Include="Authors\BenceLenart.cs" />
```
I just tried the feed and it seems to return data. I did however notice that the uri for the website url is missing the "h" in http.

Cheesebaron replied:
I guess he has fixed it. When I fetched the feed during my test it didn't contain any items.

nickrandolph commented on src/Firehose.Web/Authors/NigelSampson.cs:
```diff
@@ -1,33 +0,0 @@
-		public string TwitterHandle => "nigelsampson";
-		public string GitHubHandle => "nigel-sampson";
-		public string GravatarHash => "ffccec4c43820d5370ad22de8d569d77";
-		public IEnumerable<Uri> FeedUris => new [] { new Uri("https://compiledexperience.com/blog/syndicate/atom/") };
```
If we change the feed to https://compiledexperience.com/blog/syndicate/rss/ it should parse.
I suspect we may still want to exclude based on lack of xamarin content

Cheesebaron commented:
@planetxamarin/owners I've rebased the PR and removed 3 authors:

- @nigel-sampson does not have any Xamarin related posts in feed
- @achrafbenalaya does not provide proper Atom or RSS feed
- @icebeam7 does not meet performance requirements (currently set at 15 seconds timeout) most of the time it exceeds that, unless the feed is hot

Without these authors, getting all author feeds takes _8 seconds_!

Cheesebaron commented:
@jamesmontemagno how do we get this into a staging environment?

jamesmontemagno commented:
I created a branch, let's push it onto there and i can get it going

Cheesebaron commented:
@jamesmontemagno don't think I want to update to netcore in this PR, it is not just a single line change

jamesmontemagno commented:
ah, gotcha, let's just create a new branch and then i can test it out there. i thought this was something else lol.

Cheesebaron commented:
The branch is on this repo now too.

jamesmontemagno commented:
do you just want to push all the code there and then open a pr
and i'll get it building
